spec/app/domain/util/socket_server_spec.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe Util::SocketService do
+
+  let(:service)  do
+    Util::SocketService.new(
+      socket: socket
+    )
+  end
+
+  describe('#run') do
+    context 'When a socket is started and the socket file already exists' do
```

Comment:
All the `context` blocks that start with "when a socket is started" should go into separate blocks, like:
```ruby
context 'when a socket is started' do
  context 'when the socket file already exists' do
    it 'raises an error' do
      ...
    end
  end
  context 'when the directory can not be found' do
    ...
   end
end
```